index.bs

```diff
@@ -1074,6 +1085,221 @@ spec: webidl
+    document defines the following <a>errors</a>, algorithms, and <a>extension
+    commands</a> for the [[WebDriver]] specification.
+  </p>
+
+  The <dfn>no such request</dfn> <a>error</a> has a [=response/status=] of 404
+  and an a JSON error code of `"no such request"`.
+
+  <div algorithm="detect-pending-request">
+    <p>
+      To <dfn lt="detect a pending request">detect a pending request</dfn>
+      for a JSON object |permissionDesc| and string |type|, the UA must perform
+      the following steps.
+    </p>
+    <ol>
+      <li>Let |rootDesc| be the object |permissionDesc| refers to, <a>converted
+      to an IDL value</a> of type {{PermissionDescriptor}}. If this throws an
+      exception, return <a>error</a> with <a>error code</a> <a>invalid
+      argument</a>.</li>
+      <li>Let |typedDescriptor| be the object |permissionDesc| refers to,
+      <a>converted to an IDL value</a> of
+      <code>|rootDesc|.{{PermissionDescriptor/name}}</code>'s <a>permission
+      descriptor type</a>. If this throws an exception, return <a>error</a>
+      with <a>error code</a> <a>invalid argument</a>.</li>
+      <li>If |type| is "`use`":
+        <ol>
+          <li>*EachUseRequest*: For each <a>request permission to use</a>
+            operation that is awaiting user input:
+            <ol>
+              <li>Let |pendingDescriptor| be the {{PermissionDescriptor}}
+                associated with the request.</li>
+              <li>[=map/For each=] |key| → |value| of |pendingDescriptor|:
+                <ol>
+                  <li>If |typedDescriptor|[|key|] does not equal |value|, continue
+                  to the next iteration of the loop labeled
+                  *EachUseRequest*.</li>
+                </ol>
+              </li>
+              <li>Return true.</li>
+            </ol>
+          </li>
+        </ol>
+      </li>
+      <li>Else:
+        <ol>
+          <li class="assertion">Assert: |type| is "`choose`".
+          <li>*EachChooseRequest*: For each <a>prompt the user to choose</a>
+          operation that is awaiting user input:
+            <ol>
+              <li>Let |pendingDescriptor| be the {{PermissionDescriptor}}
+                associated with the request.</li>
+              <li>[=map/For each=] |key| → |value| of |pendingDescriptor|:
+                <ol>
+                  <li>If |typedDescriptor|[|key|] does not equal |value|, continue
+                  to the next iteration of the loop labeled
+                  *EachChooseRequest*.</li>
+                </ol>
+              </li>
+              <li>Return true.</li>
+            </ol>
+          </li>
+        </ol>
+      </li>
+      <li>Return false.
+    </ol>
+  </div>
+
+  <section>
+    <h3 id="grant-command">
+      Grant
+    </h3>
+    <table>
+      <tbody>
+        <tr>
+          <th>HTTP Method</th>
+          <th><a lt="extension command prefix">Prefix</a></th>
```

Comment:
Some of the text is definitely slanted towards vendor-specific extensions, but the use case for web standards is more clear in [the "Extensions" sub-section of "Design Notes"](https://w3c.github.io/webdriver/webdriver-spec.html#extensions):

> WebDriver aims to allow other standards to provide extensions to support new
> functionality, and to allow conformance tests that cannot be implemented
> entirely in ECMAScript to be written in a vendor-neutral way. 

I was also a little confused about URL/URI specification while writing this. Earlier today, I opened a couple pull requests that I hope will clear things up a bit:

- https://github.com/w3c/webdriver/pull/952
- https://github.com/w3c/webdriver/pull/953